mempool/mempool.go

```diff
@@ -448,7 +459,8 @@ func (mem *Mempool) ReapMaxTxs(max int) types.Txs {
 // Update informs the mempool that the given txs were committed and can be discarded.
 // NOTE: this should be called *after* block is committed by consensus.
 // NOTE: unsafe; Lock/Unlock must be managed by caller
-func (mem *Mempool) Update(height int64, txs types.Txs, filter func(types.Tx) bool) error {
+func (mem *Mempool) Update(height int64, txs types.Txs,
+	preCheckFilter func(types.Tx) bool, postCheckFilter func(types.Tx, *abci.ResponseCheckTx) bool) error {
```

Comment:
Why are those filters passed to the `Update` method directly? It feels odd to have two places to parametrise.

Reply:
The idea behind passing these to the Update method directly was that we need a way to live-update the current filters used within the mempool. The reason being, if the consensus param changes, we want the mempool to update immediately. Are you suggesting having a separate `mempool.UpdateFilters(prefilter, postfilter)` method?

Reply:
An alternative might be for the filters to have access to something that can load the latest ConsensusParams. Then we only need to set them once and they can read from the canonical shared state where the latest ConsensusParams are found

Reply:
I think that is a better alternative. I'd opt to keep the update signature as is within this PR, and plan to do what you're proposing soon. (It would likely entail either making consensus params a pointer, or adding getLatestConsensusParams method somewhere)

Reply:
In favour of @ebuchman approach.